Dockerfile

```diff
@@ -5,42 +5,66 @@ FROM timescale/timescaledb:${TIMESCALEDB_VERSION}-${PG_VERSION_TAG} as builder
 MAINTAINER Timescale https://www.timescale.com
 ARG PG_VERSION_TAG
 
-RUN set -ex \
-    && apk add --no-cache --virtual .build-deps \
-                coreutils \
-                dpkg-dev dpkg \
-                gcc \
-                libc-dev \
-                make \
-                util-linux-dev \
-                clang \
-		llvm \
-                git \
-                llvm-dev clang-libs \
-		bison \
-		dpkg-dev dpkg \
-		flex \
-		gcc \
-		libc-dev \
-		libedit-dev \
-		libxml2-dev \
-		libxslt-dev \
-		linux-headers \
-	 	clang g++ \
-		make \
-		openssl-dev \
-		perl-utils \
-		perl-ipc-run \
-		util-linux-dev \
-		zlib-dev \
-		icu-dev
+RUN set -ex; \
+    apk add --no-cache --virtual .build-deps \
+        coreutils \
+        dpkg-dev dpkg \
+        gcc \
+        libc-dev \
+        make \
+        util-linux-dev \
+        clang \
+        llvm \
+        git \
+        llvm-dev clang-libs \
+        bison \
+        dpkg-dev dpkg \
+        flex \
+        gcc \
+        libc-dev \
+        libedit-dev \
+        libxml2-dev \
+        libxslt-dev \
+        linux-headers \
+        clang g++ \
+        make \
+        openssl-dev \
+        perl-utils \
+        perl-ipc-run \
+        util-linux-dev \
+        zlib-dev \
+        icu-dev
 
-ENV RUSTUP_HOME=/usr/local/rustup \
-    CARGO_HOME=/usr/local/cargo \
-    PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.53.0 \
-    # prevents some issues with openssl
-    RUSTFLAGS="-C target-feature=-crt-static"
+RUN set -ex; \
+    git clone  --branch v1.12 --depth 1 \
+         https://github.com/dimitri/pgextwlist.git /pgextwlist; \
+    cd /pgextwlist; \
+    make; \
+    make install; \
+    mkdir `pg_config --pkglibdir`/plugins; \
+    cp /pgextwlist/pgextwlist.so `pg_config --pkglibdir`/plugins; \
+    rm -rf /pgextwlist
+
+RUN set -eux; \
+    SCCACHE_VERSION=v0.2.15; \
+    APK_ARCH="$(apk --print-arch)"; \
+    SCCACHE_FILE="sccache-${SCCACHE_VERSION}-${APK_ARCH}-unknown-linux-musl" \
+    SCCACHE_ARCHIVE="${SCCACHE_FILE}.tar.gz"; \
+    SCCACHE_URL="https://github.com/mozilla/sccache/releases/download/${SCCACHE_VERSION}/${SCCACHE_ARCHIVE}"; \
+    wget "${SCCACHE_URL}"; \
+    tar -xf ${SCCACHE_ARCHIVE}; \
+    cp ${SCCACHE_FILE}/sccache /usr/local/bin; \
+    chmod +x /usr/local/bin/sccache
+ENV RUSTC_WRAPPER=sccache
+
+USER postgres
```

Comment:
Previously we were installing rust as root and then chmodding the rust install etc. It just seemed weirdly fragile to me, and I had some issues when trying to debug stuff in the half-built container (which I can't remember exactly anymore 😅 ). That's why I refactored this to install rust as postgres.

Reply:
So it turns out that the issue is that `initdb` cannot be run as root, which I forgot about, so that means you need to run `cargo pgx init` as another user (i.e. `postgres`), but if you wish to do this without changing anything else, there are problems:

* Rustup is intended to be run by each user on the system that wants Rust, and many paths are relative to the current user's `HOME`, e.g. `CARGO_HOME`, `RUSTUP_HOME`, and `PGX_HOME`. 
* If you just change `USER` in the Dockerfile, `HOME` doesn't change, and the permissions will all be that of whatever user installed Rust (root in this case).

In the base image we're using, `HOME` is set to `/root`, and `HOME` for the `postgres` user is set to `/var/lib/postgresql` - but `USER postgres` doesn't change the value of `HOME` for subsequent `RUN` commands, it only takes effect when running the final image. As a result, we either need to set `HOME` to something; install things in `/var/lib/postgresql` (which feels gross); or create a directory we can use like `HOME` using the various `*_HOME` env vars that Rust and friends respect. I chose the first one for simplicity.

Reply:
👍